### PR TITLE
replace ClojureScript with self-hosted version

### PR DESCRIPTION
**Pull request by roman01la**
This PR replaces current ClojureScript support via server-side compiler with self-hosted version, [see](http://swannodette.github.io/2015/07/29/clojurescript-17/).

### CODE REVIEW

File: public/js/processors/processor.js [hunk 1]
```diff
@@ -583,52 +583,58 @@ var processors = jsbin.processors = (function () {
...
+
+          // init worker
+          if (window.Worker) {
+            initReady = ready;
+            worker = new Worker(jsbin.static + '/js/workers/cljs-worker.js');
+            worker.addEventListener('error', rejectWorker, false);
+            worker.addEventListener('message', workerReady, false);
+            worker.postMessage({ name: 'cljs', path: jsbin.static + '/js/vendor/cljs.js' });
+          } else {
+            console.error('Web Workers API is not supported http://caniuse.com/#feat=webworkers');
```
Comment (remy) [inline public/js/processors/processor.js]: You might need to be more invasive with this - i.e. an alert...

**Comment by remy** [comment]
Does this PR also mean that it can be added to the full output support, via https://github.com/jsbin/pennyworth?

**Comment by roman01la** [comment]
It shouldn't be hard to add this to server-side, but why would JS Bin need this on server?

**Comment by roman01la** [comment]
Btw, is it possible to use sandbox from client-side processor (`processor.js`)? I could use it instead of Web Worker to support lte IE9. Another question is if someone really wants to run CLJS in IE9 and below.

**Comment by remy** [comment]
Re: server side support: if you remove the edit link, the code should still render.

Re: sandbox. I'm not really sure. I think the sandbox is destroyed over and over, though there is a method to get the current context IIRC

**Comment by roman01la** [comment]
Makes sense. I'll prepare another PR for jsbin/pennyworth. Are you ok with Web Worker on the client-side or should I try iframe?

**Comment by remy** [comment]
Worker works for me.

**Comment by roman01la** [comment]
How can I test pennyworth with local JS Bin? Can't figure out how it works...

**Comment by remy** [comment]
If you've got pennyworth running, you need to add this to your local jsbin config to get it to point to pennyworth:

```json
  "processor": {
    "address": 5555
  },
```

Or you should be able to just create tests (based on existing tests) that shows working and failing code.

**Comment by roman01la** [comment]
Done! I think that's it.

**Pull request merged**